RiotSwiftUI/Modules/Settings/ChangePassword/Test/UI/ChangePasswordUITests.swift

```diff
@@ -0,0 +1,140 @@
+// limitations under the License.
+//
+
+import XCTest
+import RiotSwiftUI
+
+class ChangePasswordUITests: MockScreenTest {
+
+    override class var screenType: MockScreenState.Type {
+        return MockChangePasswordScreenState.self
+    }
+
+    override class func createTest() -> MockScreenTest {
+        return ChangePasswordUITests(selector: #selector(verifyChangePasswordScreen))
+    }
+
+    func verifyChangePasswordScreen() throws {
+        guard let screenState = screenState as? MockChangePasswordScreenState else { fatalError("no screen") }
+        switch screenState {
+        case .allEmpty:
+            verifyAllEmpty()
+        case .cannotSubmit:
+            verifyCannotSubmit()
+        case .canSubmit:
+            verifyCanSubmit()
+        case .canSubmitAndSignoutAllDevicesChecked:
+            verifyCanSubmitAndSignoutAllDevicesChecked()
+        }
+    }
+    
+    func verifyAllEmpty() {
+        XCTAssertTrue(app.staticTexts["titleLabel"].exists, "The title should be shown.")
+        
+        let oldPasswordTextField = app.secureTextFields["oldPasswordTextField"]
+        XCTAssertTrue(oldPasswordTextField.exists, "The text field should be shown.")
+        XCTAssertEqual(oldPasswordTextField.label, "old password", "The text field should be showing the placeholder before text is input.")
+
+        let newPasswordTextField1 = app.secureTextFields["newPasswordTextField1"]
+        XCTAssertTrue(newPasswordTextField1.exists, "The text field should be shown.")
+        XCTAssertEqual(newPasswordTextField1.label, "new password", "The text field should be showing the placeholder before text is input.")
+
+        let newPasswordTextField2 = app.secureTextFields["newPasswordTextField2"]
+        XCTAssertTrue(newPasswordTextField2.exists, "The text field should be shown.")
+        XCTAssertEqual(newPasswordTextField2.label, "confirm password", "The text field should be showing the placeholder before text is input.")
+        
+        let submitButton = app.buttons["submitButton"]
+        XCTAssertTrue(submitButton.exists, "The submit button should be shown.")
+        XCTAssertFalse(submitButton.isEnabled, "The submit button should be disabled when not able to submit.")
+
+        let signoutAllDevicesToggle = app.switches["signoutAllDevicesToggle"]
+        XCTAssertTrue(signoutAllDevicesToggle.exists, "Sign out all devices toggle should exist")
+        XCTAssertTrue(signoutAllDevicesToggle.isOn, "Sign out all devices should be checked")
+    }
+    
+    func verifyCannotSubmit() {
+        XCTAssertTrue(app.staticTexts["titleLabel"].exists, "The title should be shown.")
+
+        let oldPasswordTextField = app.secureTextFields["oldPasswordTextField"]
+        XCTAssertTrue(oldPasswordTextField.exists, "The text field should be shown.")
+        XCTAssertEqual(oldPasswordTextField.value as? String, "••••••••", "The text field should be showing the placeholder before text is input.")
```

Comment:
The message looks like a copy-paste from the previous test (I can't wait for screenshot testing instead of all of these!)

Reply:
🤦‍♂️